### PR TITLE
Sync apptags in orderforms to status-db

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds a CLI command to sync orderform apptags to the status database

Contacts: @emiliaol @mayabrandi @vwirta @patrikgrenfeldt

### How to test informational usage:
0. install on stage (add instructions on how to as needed)
1. go to hasta 
1. us
1. download an rml or microbial orderform 
1. run `cg sync apptag [OPTIONS] EXCEL_PATH PREP_CATEGORY SIGNATURE` (e.g. `cg sync application ~/1603-8_1603-8_Orderform_Microbial_WGS.xlsx mic PG`)

**Expected outcome:**
Information on which tags that could be archived or activated. Info about rollback

### How to test activational usage:
1. run `cg sync apptag [OPTIONS] EXCEL_PATH PREP_CATEGORY SIGNATURE --activate` (e.g. `cg sync application ~/1603-8_1603-8_Orderform_Microbial_WGS.xlsx mic PG --activate`)

**Expected outcome:**
Information on which tags that could be archived. Info on which tags were activated. Info about committing to database.

### How to test inactivational usage:
1. run `cg sync application [OPTIONS] EXCEL_PATH PREP_CATEGORY SIGNATURE --inactivate` (e.g. `cg sync application ~/1603-8_1603-8_Orderform_Microbial_WGS.xlsx mic PG --inactivate`)

**Expected outcome:**
Information on which tags that could be activated. Info on which tags were inactivated. Info about committing to database.

- [x] Code reviewed by @ingkebil  
- [x] Tested by @patrikgrenfeldt 
- [x] Deploy OK:d by @patrikgrenfeldt 

This is a minor **version bump** because it expands the CLI in a backwards compatible manner

### CODE REVIEW

**Comment by patrikgrenfeldt** [comment]
informational usage:
image

activational usage:
image

inactivational usage:
image

👍

**Comment by ingkebil** [comment]
The tests fail ;)

**Comment by patrikgrenfeldt** [comment]
> The tests fail ;)

They actually do fail :) It will be so awesome when we have a clear distinction between lint-failures and test-failures

**Comment by patrikgrenfeldt** [comment]
> The tests fail ;)

They're pass again 🗡

File: cg/cli/sync.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from cg.store import Store
+
+from cg.store.api.sync import sync_apptags
+
+LOG = logging.getLogger(__name__)
+
+
+@click.group()
+@click.pass_context
+def sync(context):
```
Comment (patrikgrenfeldt) [inline cg/cli/sync.py]: suggest move to cli.import_cmd!

File: cg/store/api/sync.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
+"""Sync functionality"""
```
Comment (patrikgrenfeldt) [inline cg/store/api/sync.py]: suggest move to import part of API

File: cg/cli/sync.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                                                                   'to find the applications')
+@click.option('-n', '--tag-column', default=2, help='(optional) zero-based column where to find '
+                                                    'the applications')
+@click.option('-a', '--activate', is_flag=True, help='Activate archived tags found in the '
+                                                     'orderform')
+@click.option('-i', '--inactivate', is_flag=True, help='Inactivate tags not found in the orderform')
+@click.pass_context
+def apptag(context, excel_path, prep_category, signature, sheet_name, tag_column, activate,
+           inactivate):
+    """Sync apptags in the status-db"""
```
Comment (patrikgrenfeldt) [inline cg/cli/sync.py]:
```suggestion
    """Sync apptags in the status-db, will run as dry-run by default"""
```

File: cg/store/api/sync.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import sys
+from datetime import datetime
+
+import xlrd
+from cg.store import Store
+
+
+def sync_apptags(store: Store, excel_path: str, prep_category: str, sign: str, sheet_name:
+                 str, tag_column: int, activate: bool, inactivate: bool):
+    """
```
Comment (patrikgrenfeldt) [inline cg/store/api/sync.py]: suggest move to cli where it can be seen

File: cg/store/api/sync.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+import logging
+import sys
+from datetime import datetime
+
+import xlrd
+from cg.store import Store
+
+
+def sync_apptags(store: Store, excel_path: str, prep_category: str, sign: str, sheet_name:
```
Comment (patrikgrenfeldt) [inline cg/store/api/sync.py]: sign could be automatically retrieved

File: cg/cli/sync.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/cli/sync.py]: done!

File: cg/store/api/sync.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/store/api/sync.py]: done

File: cg/store/api/sync.py (hunk 4 above)
Comment (patrikgrenfeldt) [reply cg/store/api/sync.py]: done

File: cg/store/api/sync.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+    if not activate and not inactivate:
+        logging.info('no change mode requested, rolling back transaction')
+        store.rollback()
+    else:
+        logging.info('all applications successfully synced, committing transaction')
+        store.commit()
+
+
+def get_tag_from_raw_data(raw_data, column: int):
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/store/api/sync.py]: done

File: cg/store/api/sync.py [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    orderform_application_tags = []
+
+    for raw_row in get_raw_data_from_xl(excel_path, sheet_name, tag_column):
+        tag = get_tag_from_raw_data(raw_row, tag_column)
+        logging.info('Found: %s in orderform', tag)
+        orderform_application_tags.append(tag)
+
+    if not orderform_application_tags:
+        logging.error('No applications found in column %s (zero-based), exiting', tag_column)
+        sys.exit()
```
[the remark this answers is not in the document]
Comment (patrikgrenfeldt) [reply cg/store/api/sync.py]: I raise instead, what do you think of it? @ingkebil

[the remark this answers is not in the document]
Comment (ingkebil) [reply cg/store/api/sync.py]: yes, that's the way to go

**Comment by patrikgrenfeldt** [comment]
> 🦅

@ingkebil I put you as reviewed by, that's what you meant, these changes fly, right ;)

**Comment by patrikgrenfeldt** [comment]
deployed

**Pull request merged**